### PR TITLE
transition to single Makefile

### PR DESCRIPTION
**Pull request by ericvw**
After inspiration from the "Recursive Make Considered Harmful" paper,
this change collapses the recursive Makefiles into a single Makefile.
Since the project is fairly small and constrained, work to split out the
Makefile into smaller fragments and utilizing make's `include` directive
has been deferred.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -49,33 +62,36 @@ build: $(SRCS_JS)
...
+$(TSLINT_TEST_TARGET): $(SRCS_TEST_TS) $(TSLINT_CONFIG) $(SRCS_TSLINT_RULES_JS)
 	@$(RM) $(TSLINT_TEST_TARGET)
-	@$(TSLINT) $(foreach file,$(SRCS_TEST_TS),-f $(file))
+	@$(TSLINT) -r $(TSLINT_RULES_DIR) $(foreach file,$(SRCS_TEST_TS),-f $(file))
 	@$(TOUCH) $(TSLINT_TEST_TARGET)
 
-%.js: %.ts
-	@$(TSC) $<
+$(SRCS_TSLINT_RULES_JS): $(SRCS_TSLINT_RULES_TS)
+	@$(TSC) $(TSLINT_D) $<
```
Comment (thmiceli) [inline Makefile]: We can use $(SRCS_TSLINT_RULES_TS) instead of $< for consistency with the rules below.

**Comment by thmiceli** [comment]
LGTM beside the minor concern comment I made.

**Comment by ericvw** [comment]
Updated PR based on feedback.

**Comment by ericlu88** [comment]
LGTM

**Comment by thmiceli** [comment]
LGTM

**Pull request merged**